drivers/bluetooth/hci/userchan.c

```diff
@@ -0,0 +1,214 @@
+/* userchan.c - HCI User Channel based Bluetooth driver */
+
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <init.h>
+#include <misc/util.h>
+#include <misc/byteorder.h>
+
+#include <errno.h>
+#include <stddef.h>
+#include <poll.h>
+#include <errno.h>
+#include <sys/socket.h>
+#include <string.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/hci_driver.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#include "common/log.h"
+
+#define BTPROTO_HCI      1
+struct sockaddr_hci {
+	sa_family_t     hci_family;
+	unsigned short  hci_dev;
+	unsigned short  hci_channel;
+};
+#define HCI_CHANNEL_USER 1
+
+#define SOL_HCI          0
+
+#define H4_NONE          0x00
+#define H4_CMD           0x01
+#define H4_ACL           0x02
+#define H4_SCO           0x03
+#define H4_EVT           0x04
+
+static BT_STACK_NOINIT(rx_thread_stack, CONFIG_BT_RX_STACK_SIZE);
```

Comment:
You only need CONFIG_ARCH_POSIX_RECOMMENDED_STACK_SIZE.
The underlying (host) pthread has another stack. This zephyr stack does not hold the real thread stack.

Reply:
Thanks. I'll fix this in the next revision.